src/Microsoft.ML.TimeSeries/SRCNNAnomalyDetector.cs

```diff
@@ -0,0 +1,237 @@
+            return new VersionInfo(
+                modelSignature: "SRCNTRNS",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(SrCnnAnomalyDetector).Assembly.FullName);
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, Options options, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(options, nameof(options));
+            env.CheckValue(input, nameof(input));
+
+            return new SrCnnAnomalyDetector(env, options).MakeDataTransform(input);
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(ctx, nameof(ctx));
+            env.CheckValue(input, nameof(input));
+
+            return new SrCnnAnomalyDetector(env, ctx).MakeDataTransform(input);
+        }
+
+        private static SrCnnAnomalyDetector Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new SrCnnAnomalyDetector(env, ctx);
+        }
+
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, DataViewSchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        IStatefulTransformer IStatefulTransformer.Clone()
+        {
+            var clone = (SrCnnAnomalyDetector)MemberwiseClone();
+            clone.InternalTransform.StateRef = (SrCnnAnomalyDetectionBaseCore.State)clone.InternalTransform.StateRef.Clone();
+            clone.InternalTransform.StateRef.InitState(clone.InternalTransform, InternalTransform.Host);
+            return clone;
+        }
+
+        internal SrCnnAnomalyDetector(IHostEnvironment env, Options options)
+            : base(new SrCnnArgument(options), LoaderSignature, env)
+        {
+        }
+
+        internal SrCnnAnomalyDetector(IHostEnvironment env, ModelLoadContext ctx)
+            : base(env, ctx, LoaderSignature)
+        {
+        }
+
+        private SrCnnAnomalyDetector(IHostEnvironment env, SrCnnAnomalyDetector transform)
+           : base(new SrCnnArgument(transform), LoaderSignature, env)
+        {
+        }
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            InternalTransform.Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // <base>
+            base.SaveModel(ctx);
+        }
+    }
+
+    /// <summary>
+    /// Detect anomalies in time series using Spectral Residual
+    /// </summary>
```

Comment:
In general, a public C# API may need more details than this line. For example, [https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.matrixfactorizationtrainer?view=ml-dotnet-preview](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.matrixfactorizationtrainer?view=ml-dotnet-preview) and [https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.sdcanoncalibratedbinarytrainer?view=ml-dotnet](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.sdcanoncalibratedbinarytrainer?view=ml-dotnet)
.

Reply:
Document added.